pandas/core/internals/blocks.py

```diff
@@ -1176,15 +1179,43 @@ def _interpolate_with_fill(
         # We only get here for non-ExtensionBlock
         fill_value = convert_scalar_for_putitemlike(fill_value, self.values.dtype)
 
-        values = missing.interpolate_2d(
-            values,
+        interpolate_2d = functools.partial(
+            missing.interpolate_2d,
             method=method,
-            axis=axis,
             limit=limit,
             fill_value=fill_value,
             dtype=self.dtype,
         )
 
+        if limit_area is None:
```

Comment:
can you put this in a function in `missing` and just call here?

Reply:
Okay. But if I do this I could also directly integrate it into `missing.interpolate_2d`, e.g. like [in this branch](https://github.com/pandas-dev/pandas/compare/master...cchwala:quick_test_interpolate_limit_area). Advantage, no need to introduce yet another function that does some interpolation. Disadvantage, obfuscates `missing.interpolate_2d`. For this, adding some comments would help for sure.

If you prefer the separate function in `missing`, what function name would you choose?

Reply:
> e.g. like [in this branch](https://github.com/pandas-dev/pandas/compare/master...cchwala:quick_test_interpolate_limit_area).

I would merge recursive solution here for review (so that ci has been run against it, i think needs changes to pandas/core/arrays/categorical.py see https://github.com/pandas-dev/pandas/compare/master...simonjayhawkins:fix_interpolate_limit_area_and_limit_direction_with_pad)

Reply:
Okay. Merged and CI is running.

I have also pushed [a branch](https://github.com/pandas-dev/pandas/compare/master...cchwala:quick_test_limit_area_option2) with an example implementation using a separate function in `missing`. The name of the new function is not good, but the implementation would probably look something like this.

For both solutions, tests for interpolation pass, but I did not run the full test suite. Will see what changes are required when CI is done.

Reply:
Test are green. 

If you find the recursive solution appropriate, I would add a comment to the function that briefly explains the reasoning. Anything else that needs improvement?

Reply:
@jreback